Assets/GRPC.NET/Scripts/PushPullStream.cs

```diff
@@ -2,31 +2,24 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Threading;
+using Best.HTTP;
+using Best.HTTP.Request.Upload;
 
 namespace GRPC.NET
 {
-    public class PushPullStream : Stream
+    public class PushPullStream : UploadStreamBase
```

Comment:
new base class for the uploader with handles receiving the thread signal on its own.